PowerShell/canvas-unpack-pack.ps1

```diff
@@ -132,6 +132,112 @@ function Invoke-Share-Canvas-App-with-AAD-Group
 	}
 }
 
+<#
+Read all the Canvas Apps part of the solution.
+Apply PowerShell command to bypass the Canvas app consents.
+#>
+function ByPass-Canvas-App-Consents
+{
+    param (
+        [Parameter(Mandatory)] [String]$microsoftPowerAppsAdministrationPowerShellModule,
+        [Parameter(Mandatory)] [String]$powerAppsAdminModuleVersion,
+        [Parameter(Mandatory)] [String]$tenantId,
+        [Parameter(Mandatory)] [String]$clientId,
+        [Parameter(Mandatory)] [String]$clientSecret,
+        [Parameter(Mandatory)] [String]$microsoftXrmDataPowerShellModule,
+        [Parameter(Mandatory)] [String]$XrmDataPowerShellVersion,
+        [Parameter(Mandatory)] [String]$serviceConnection,
+        [Parameter(Mandatory)] [String]$environmentId,
+        [Parameter(Mandatory)] [String]$dataverseConnectionString,
+        [Parameter(Mandatory)] [String]$solutionName
+    )
+
+    Import-Module $microsoftPowerAppsAdministrationPowerShellModule -Force -RequiredVersion $powerAppsAdminModuleVersion -ArgumentList @{ NonInteractive = $true }
+    Add-PowerAppsAccount -TenantID $tenantId -ApplicationId $clientId -ClientSecret $clientSecret
+    #$microsoftXrmDataPowerShellModule = '$(CoETools_Microsoft_Xrm_Data_PowerShell)'
+    Import-Module $microsoftXrmDataPowerShellModule -Force -RequiredVersion $XrmDataPowerShellVersion -ArgumentList @{ NonInteractive = $true }
+    $conn = Get-CrmConnection -ConnectionString "$dataverseConnectionString"
+
+    $environmentName = "$environmentId"
+
+    # Fetch all solution components
+    $solutions = Get-CrmRecords -conn $conn -EntityLogicalName solution -FilterAttribute "uniquename" -FilterOperator "eq" -FilterValue "$solutionName" -Fields solutionid
+    if($solutions.Count -gt 0) {
+        $solutionId = $solutions.CrmRecords[0].solutionid
+
+        $result = Get-CrmRecords -conn $conn -EntityLogicalName solutioncomponent -FilterAttribute "solutionid" -FilterOperator "eq" -FilterValue $solutionId -Fields objectid,componenttype
+        $solutionComponents = $result.CrmRecords
+
+        $optionSetMetadata = $null
+        foreach ($c in $solutionComponents){
+            $componentType = $c.componenttype
+            Write-Host "Componenttype - $componentType"           
+
+            # Logic to check the MDA
```

Comment:
@RajeevPentyala can we get rid of this commented code and the Get-OptionSetOptions function since it's not being used?